R/LSgameofchance.R

```diff
@@ -15,272 +15,146 @@
-                                              beta = combinations(k)[1], prob = p[2:length(p)]))
-  }
-  
-  # calculating difference between the simulated and the calculated probability of player 1 winning
-  dif <- abs(p_simulated - p_calculated)
-  return (list(p_simulated, p_calculated, dif, p_cumulative))
-}
-
-
-#l <- compare_function2(c(1,1,1), 2, c(0.2,0.4,0.4), 10)
-#l
-
-combinations <- function(k){
-  if (length(k) == 3){ # the number of players
-    #k <- c(1,2,3)
-    num <- prod(k)/k[1] #total number of sets
-    #sets <- vector(mode = "list", length = num)
-    sets <- matrix(nrow = num, ncol = length(k))
-    for (i in 0:(k[2]-1)){
-      for (j in 0:(k[3]-1)){
-        sets[i*k[3]+j+1,] <- c(k[1],i,j)
-      }
-    }
-    return(sets)
-  }else{
-    #k <- c(1,2,3,4)
-    #x1 <- sets
-    x1 <- combinations(k[1:(length(k) - 1)]) # k without k[length(k)]
-    x2 <- k[length(k)]  
-    y <- matrix(ncol = length(k), nrow = prod(k)/k[1])
-    for (i in 1:length(x1[,1])){
-      for (j in 1:k[length(k)]){
-        y[(i-1)*(k[length(k)])+j,] <- append(x1[i,],j-1)
-      }
-    }
-    return(y)
-  }
-} 
-
-##
 LSgameofchance   <- function(jaspResults, dataset, options, state = NULL){
 
-  #ready <- (length(options$variables) > 0)
+  # input values
+  nPlayers  <- length(options[["players"]])
+  probWin   <- sapply(options[["players"]], function(p)p$values[[1]])
+  xPoints   <- sapply(options[["players"]], function(p)p$values[[2]])
+  winPoints <- options[["winPoints"]]
+  nSims     <- options[["nSims"]]
   
+  # normalizing the probability of wining:
+  probWin   <- probWin / sum(probWin) # normalit
   
-  ## some transformation to match previous code
-  input <- list("n" = options$n, "k" = options$k, "t" = options$t, "p" = options$p,
-                "s" = options$s, "check" = options$check)
-  # input <- list("p" = "1,1", "k" = "1,1", "t" = 2, "s" = 100, "n"= 2)
-  k <- as.numeric(unlist(strsplit(input$k,",")))
-  p <- as.numeric(unlist(strsplit(input$p,",")))/sum(as.numeric(unlist(strsplit(input$p,","))))
   
   ## check errors
-  if(input$n != length(k))
-    JASP:::.quitAnalysis(gettextf(
-      "The number of players (%1$i) does not equal the numbers of points for each player when interrupted (%2$i). Please check the appropriate settings.",
-      input$n,
-      length(k)
-    ))
-  #need(max(k()) < input$t, paste("Warning: Player", which(k() == max(k())),
-  #                               " has already won the game. Adjust the inputs!")),
-  #.hasErrors(dataset, type = "limits",
-  #           message = paste("Warning: Player", which(k == max(k)),
-  #                           " has already won the game. Adjust the inputs!"),  
-  #           limits.target = input$t,
-  #           limits.max = max(k) - 1,
-  #           exitAnalysisIfErrors = TRUE)
+  if(nPlayers < 2)
+    .quitAnalysis(gettext("Warning: The number of players must be at least 2. Adjust the inputs!"))
```

Comment:
`.quitAnalysis` is now automatically imported and should't be called with `JASP:::` prefix